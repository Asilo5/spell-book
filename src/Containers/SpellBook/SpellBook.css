.spell-container {
  text-align: center;
  position: relative;
  top: 2em;
  width: 44%;
  margin: auto;
}

.spell-info {
   display: flex;
   flex-direction: column;
   justify-content: space-around;
   height: 12em;
}

.spell-info > h3 {
  font-family: 'Henny Penny', cursive;
  color: black;
  font-size: 4em;
}

.spell-info > h4 {
  margin-top: -2em;
  font-size: 20px;
}

.spell-info > p {
  font-size: 30px;
}
  
.favourite-wand {
   height: 8em;
   margin-left: 30em;
   transform: rotate(13deg);
   transition: linear 2s ease-in-out;
   cursor: grab;
}

.favourite-wand:hover {
  transform: rotate(-13deg);
}
  
.paper-img {
  display: block;
  margin: auto;
  margin-top: -20em;
}

.chosen-wand {
  height: 8em;
  margin-left: 30em;
  transform: rotate(110deg);
  transition: linear 2s ease-in-out;
  cursor: grab;
  border-radius: 50%;
  background-color: #ffeb00b8;
  box-shadow: 
    0 0px 1px #ffeb0045, 
    0 0 86px 19px #ffeb3b73, 
    0 0 23px 46px #ffeb3b9c;
}

.chosen-wand:hover {
  transform: rotate(125deg);
}