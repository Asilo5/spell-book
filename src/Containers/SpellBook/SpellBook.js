import React, { Component } from 'react';
import './SpellBook.css'; 
import { connect } from 'react-redux';
import { addFavourite, deleteFavourite } from '../../actions';
import { bindActionCreators } from 'redux';
import { PropTypes } from 'prop-types';

export class SpellBook extends Component {
  constructor() {
    super();
    this.state = {
      chosenFavourite : !false
    }
  }

  toggleFavourite = () => {
   const { addFavourite, deleteFavourite, favouriteSpells, _id, isFavourite} = this.props;
   this.setState({ chosenFavourite : !this.state.chosenFavourite })
   favouriteSpells.includes(_id) ? deleteFavourite(_id) : addFavourite(_id);
  } 

  render() {
    const { chosenFavourite } = this.state;
    const { spell, type, effect } = this.props;
    return (
        <section className='spells' > 
            <div className='spell-container'>
              <img onClick={() => this.toggleFavourite()} className={chosenFavourite ? 'favourite-wand' : 'chosen-wand'} src='https://gamepress.gg/wizardsunite/sites/wizardsunite/files/2019-04/Luna%20Lovegood-foundable_0.png' alt='luna lovegood wand to favourite spell card' />
              <div className='spell-info'>
                <h3>{spell}</h3>
                <h4>{type}</h4>
                <p>This {type} {effect}</p>
              </div>
            </div>
            <img className='paper-img' src='https://www.artsclip.com/background/XZHaFAgbeP-vintage-paper-scroll-for-short-news-slides.png' alt='background paper for spells' />
        </section>
    )
  }
}

export const mapStateToProps = ({userHouse, nameOfWizard, selectedSpell, spellBook, favouriteSpells }) => ({
  userHouse,
  nameOfWizard,
  selectedSpell,
  spellBook,
  favouriteSpells
})

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({ 
    addFavourite,
    deleteFavourite
  }, dispatch)
)

SpellBook.propTypes = {
  userHouse: PropTypes.string,
  nameOfWizard: PropTypes.string,
  selectedSpell: PropTypes.array,
  spellBook: PropTypes.array,
  favouriteSpells: PropTypes.array,
  addFavourite: PropTypes.func,
  deleteFavourite: PropTypes.func

};

export default connect(mapStateToProps, mapDispatchToProps)(SpellBook);
