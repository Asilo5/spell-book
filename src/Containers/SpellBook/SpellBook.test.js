import React from 'react';
import { shallow } from 'enzyme';
import { SpellBook, mapStateToProps, mapDispatchToProps } from './SpellBook';

describe('SpellBook', () => {
 
    describe('Spells', () => {
        let mockFavouriteSpell = [
            { _id: '5b74f3653228320021ab628b',
            spell: 'Lumos',
            type: 'Spell',
            effect: 'creates light at wand tip',
            __v: 0 }
           ];
    
           let mockSpellBook = [
            { _id: '5b74f3583228320021ab628a',
            spell: 'Locomotor Mortis',
            type: 'Curse',
            effect: 'locks opponent\'s legs',
            __v: 0 },
          { _id: '5b74f3653228320021ab628b',
            spell: 'Lumos',
            type: 'Spell',
            effect: 'creates light at wand tip',
            __v: 0 } 
           ];

        let wrapper = shallow(<SpellBook 
            userHouse='Hufflepuff'
            nameOfWizard='Consuelo'
            selectedSpell='Alohomora'
            spellBook={mockFavouriteSpell}
            favouriteSpells={mockSpellBook}
            addFavourite='2343'
            deleteFavourite='3443'
         />)

         it('should match snapshot with all information passing in correctly', () => {
             expect(wrapper).toMatchSnapshot();
         })
    })

    describe('mapStateToProps', () => {
        it('should return all state with their properties', () => {
            const mockState = {
                "userHouse": 'Hufflepuff',
                "nameOfWizard": 'Consuelo Weasley',
                "selectedSpell": 'Alohomora',
                "favouriteSpells": [   
                 { _id: '5b74f3653228320021ab628b',
                    spell: 'Lumos',
                    type: 'Spell',
                    effect: 'creates light at wand tip',
                    __v: 0 }
                 ],
                "spellBook": [
                   { _id: '5b74f3653228320021ab628b',
                   spell: 'Lumos',
                   type: 'Spell',
                   effect: 'creates light at wand tip',
                 __v: 0 }
                ],
             }
             const expected = {
                "userHouse": 'Hufflepuff',
                "nameOfWizard": 'Consuelo Weasley',
                "selectedSpell": 'Alohomora',
                "favouriteSpells": [
                 { _id: '5b74f3653228320021ab628b',
                 spell: 'Lumos',
                 type: 'Spell',
                 effect: 'creates light at wand tip',
                 __v: 0 }
                ],
                 "spellBook": [
                     { _id: '5b74f3653228320021ab628b',
                    spell: 'Lumos',
                    type: 'Spell',
                    effect: 'creates light at wand tip',
                    __v: 0 }
                 ],
             }
             const mappedProps = mapStateToProps(mockState);
             expect(mappedProps).toEqual(expected);
        })
    })

    describe('mapDispatchToProps', () => {
        
    })

})