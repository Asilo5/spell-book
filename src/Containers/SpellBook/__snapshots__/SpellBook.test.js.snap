// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellBook Spells should match snapshot with all information passing in correctly 1`] = `
<section
  className="spells"
>
  <div
    className="spell-container"
  >
    <img
      alt="luna lovegood wand to favourite spell card"
      className="favourite-wand"
      onClick={[Function]}
      src="https://gamepress.gg/wizardsunite/sites/wizardsunite/files/2019-04/Luna%20Lovegood-foundable_0.png"
    />
    <div
      className="spell-info"
    >
      <h3 />
      <h4 />
      <p>
        This 
         
      </p>
    </div>
  </div>
  <img
    alt="background paper for spells"
    className="paper-img"
    src="https://www.artsclip.com/background/XZHaFAgbeP-vintage-paper-scroll-for-short-news-slides.png"
  />
</section>
`;
